pycaret/internal/plots/time_series.py

```diff
@@ -910,9 +910,9 @@ def plot_time_series_decomposition(
         return fig, return_data_dict
 
     data_kwargs = data_kwargs or {}
-    fig_kwargs = fig_kwargs or {}
-
+    period = data_kwargs.get("seasonal_period")
```

Comment:
It will be (since I am passing it from outside in my code). Users do not access this function directly.

Reply:
`    data_kwargs = data_kwargs or {}` this line should be removed then

Reply:
I have added checks for this now instead. That is better since I anyway needed to do so for another Issue. I will close this out once the tests pass. If you have more comments on this, I will incorporate them in a future PR.